test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -748,25 +748,38 @@ private static void deleteAllNodeShutdownMetadata() throws IOException {
     }
 
     protected static void wipeAllIndices() throws IOException {
+        final String[] indicesToIgnore = new String[] {
+            // ignore ilm history which can pop up after deleting all data streams but shouldn't interfere
+            ".ds-ilm-history-*"
+        };
+
         boolean includeHidden = minimumNodeVersion().onOrAfter(Version.V_7_7_0);
         try {
-            //remove all indices except ilm history which can pop up after deleting all data streams but shouldn't interfere
-            final Request deleteRequest = new Request("DELETE", "*,-.ds-ilm-history-*");
+            final Request deleteRequest = new Request("DELETE", "*,-" + String.join(",-", indicesToIgnore));
             deleteRequest.addParameter("expand_wildcards", "open,closed" + (includeHidden ? ",hidden" : ""));
-            RequestOptions allowSystemIndexAccessWarningOptions = RequestOptions.DEFAULT.toBuilder()
-                .setWarningsHandler(warnings -> {
-                    if (warnings.size() == 0) {
-                        return false;
-                    } else if (warnings.size() > 1) {
-                        return true;
-                    }
-                    // We don't know exactly which indices we're cleaning up in advance, so just accept all system index access warnings.
-                    final String warning = warnings.get(0);
-                    final boolean isSystemIndexWarning = warning.contains("this request accesses system indices")
-                        && warning.contains("but in a future major version, direct access to system indices will be prevented by default");
-                    return isSystemIndexWarning == false;
-                }).build();
-            deleteRequest.setOptions(allowSystemIndexAccessWarningOptions);
+
+            if (minimumNodeVersion().onOrAfter(Version.CURRENT)) {
```

Comment:
`Version.CURRENT` will change when we cut a new release, so this branch will only ever be taken in non-mixed clusters. Shouldn't we check against the version the feature reset API was implemented? (or whatever version it's functional enough to be usable in at least)

Reply:
That's totally correct. I used `Version.CURRENT` in order to delay thinking about BWC issues and merging, but I didn't even put a TODO in, so it would have been easy to forget.

- [ ] put in a definite version, or a clear BWC todo comment